packages/cron-job/Entity/CronJobExecution.php

```diff
@@ -0,0 +1,184 @@
+            inversedBy: 'executions',
+        ),
+        ORM\JoinColumn(
+            name: 'cron_job_id',
+            referencedColumnName: 'id',
+            nullable: false,
+            onDelete: 'CASCADE',
+        )
+    ]
+    private ?CronJob $cronJob = null;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getRequestedAt(): ?\DateTimeImmutable
+    {
+        return $this->requestedAt;
+    }
+
+    public function setRequestedAt(?\DateTimeImmutable $requestedAt): self
+    {
+        $this->requestedAt = $requestedAt;
+
+        return $this;
+    }
+
+    public function isForce(): bool
+    {
+        return $this->force;
+    }
+
+    public function setForce(bool $force): self
+    {
+        $this->force = $force;
+
+        return $this;
+    }
+
+    public function getExecutionStartedAt(): ?\DateTimeImmutable
+    {
+        return $this->executionStartedAt;
+    }
+
+    public function setExecutionStartedAt(?\DateTimeImmutable $executionStartedAt): self
+    {
+        $this->executionStartedAt = $executionStartedAt;
+
+        return $this;
+    }
+
+    public function getExecutionEndedAt(): ?\DateTimeImmutable
+    {
+        return $this->executionEndedAt;
+    }
+
+    public function setExecutionEndedAt(?\DateTimeImmutable $executionEndedAt): self
+    {
+        $this->executionEndedAt = $executionEndedAt;
+
+        return $this;
+    }
+
+    public function getExecutionDelay(): ?int
+    {
+        return $this->executionDelay;
+    }
+
+    public function setExecutionDelay(?int $executionDelay): self
+    {
+        $this->executionDelay = $executionDelay;
+
+        return $this;
+    }
+
+    public function getExitCode(): ?int
+    {
+        return $this->exitCode;
+    }
+
+    public function setExitCode(?int $exitCode): self
+    {
+        $this->exitCode = $exitCode;
+
+        return $this;
+    }
+
+    public function getError(): ?array
+    {
+        return $this->error;
+    }
+
+    public function setError(?array $error): self
+    {
+        $this->error = $error;
+
+        return $this;
+    }
+
+    public function getCronJob(): ?CronJob
+    {
+        return $this->cronJob;
+    }
+
+    public function setCronJob(?CronJob $cronJob): self
+    {
+        $this->cronJob = $cronJob;
+
+        return $this;
+    }
+
+    public function start(): void
+    {
+        $this
+            ->setExecutionStartedAt(new \DateTimeImmutable())
+            ->setExecutionEndedAt(null);
+    }
+
+    public function end(): void
+    {
+        $this
+            ->setExitCode(0)
+            ->setExecutionEndedAt($executionEndedAt = new \DateTimeImmutable())
+            ->setExecutionDelay($executionEndedAt->getTimestamp() - $this->getExecutionStartedAt()->getTimestamp());
+    }
+
+    public function fail(?int $exitCode, ?array $error): void
+    {
+        $this
+            ->setExitCode($exitCode)
+            ->setError($error);
```

Comment:
you still need to set the execution end at and delay

Reply:
Still missing

Reply:
`$this->end()` is called before. It sets both `executionEndDate` and `executionDelay`.